includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -82,13 +83,69 @@ public function get_selector_conversion_mapping() {
 		];
 	}
 
+	/**
+	 * Mark node and it's child as being PX-verified.
+	 *
+	 * @param DOMNode $node Node.
+	 *
+	 * @return void
+	 */
+	private static function mark_node_as_px_verified_recursively( DOMNode $node ) {
+
+		ValidationExemption::mark_node_as_px_verified( $node );
+
+		if ( $node->hasChildNodes() ) {
+			foreach ( $node->childNodes as $child_node ) {
+				self::mark_node_as_px_verified_recursively( $child_node );
+			}
+		}
+	}
+
+	/**
+	 * Convert picture element into image element or mark as px verified.
+	 *
+	 * @return void
+	 */
+	protected function process_picture_elements() {
+
+		$nodes     = $this->dom->getElementsByTagName( Tag::PICTURE );
+		$num_nodes = $nodes->length;
+
+		if ( 0 === $num_nodes ) {
+			return;
+		}
+
+		for ( $index = $num_nodes - 1; $index >= 0; $index-- ) {
+
+			/** @var Element $picture_node */
+			$picture_node = $nodes->item( $index );
+
+			if ( true === $this->args['native_img_used'] ) {
+				self::mark_node_as_px_verified_recursively( $picture_node );
```

Comment:
This is a thoughtful addition. Nevertheless, I don't think we should reuse the `native_img_used` arg here. The reason is that native images will be the default soon (per #6805). Once that happens, users should get valid AMP pages and `picture` should be caught as a validation error. Therefore, I suggest there be a new argument like `allow_picture`.

Reply:
I have added a new argument option `allow_picture`.